File: cpp/example_code/ec2/terminate_instances.cpp
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/
/**
 * Before running this C++ code example, set up your development environment, including your credentials.
 *
 * For more information, see the following documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started.html
 *
 * For information on the structure of the code examples and how to build and run the examples, see
 * https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started-code-examples.html.
 *
 **/

#include <aws/core/Aws.h>
#include <aws/ec2/EC2Client.h>
#include <aws/ec2/model/TerminateInstancesRequest.h>
#include <iostream>
#include "ec2_samples.h"

//! Terminate an Amazon Elastic Compute Cloud (Amazon EC2) instance.
/*!
  \sa TerminateInstances()
  \param instanceID: An EC2 instance ID.
  \param clientConfiguration: AWS client configuration.
  \return bool: Function succeeded.
 */
bool AwsDoc::EC2::TerminateInstances(const Aws::String &instanceID,
                                     const Aws::Client::ClientConfiguration &clientConfiguration) {
// snippet-start:[cpp.example_code.ec2.TerminateInstances]
    Aws::EC2::EC2Client ec2Client(clientConfiguration);

    Aws::EC2::Model::TerminateInstancesRequest request;
    request.SetInstanceIds({instanceID});

    Aws::EC2::Model::TerminateInstancesOutcome outcome =
            ec2Client.TerminateInstances(request);
    if (outcome.IsSuccess()) {
        std::cout << "Ec2 instance '" << instanceID <<
                  "' was terminated." << std::endl;
    }
    else {
        std::cerr << "Failed to terminate ec2 instance " << instanceID <<
                  ", " <<
                  outcome.GetError().GetMessage() << std::endl;
        return false;
    }
// snippet-end:[cpp.example_code.ec2.TerminateInstances]

    return outcome.IsSuccess();
}

/*
 *
 *  main function
 *
 *  Usage: 'run_terminate_instances <instance_name>'
 *
 *  Prerequisites: An EC2 instance to terminate.
 *
 */

#ifndef TESTING_BUILD

int main(int argc, char **argv) {
    if (argc != 2) {
        std::cout << "Usage: run_terminate_instances <instance_id>"
                  << std::endl;
        return 1;
    }

    Aws::SDKOptions options;
    Aws::InitAPI(options);
    {
        Aws::Client::ClientConfiguration clientConfig;
        // Optional: Set to the AWS Region (overrides config file).
        // clientConfig.region = "us-east-1";
        Aws::String instanceID = argv[1];
        AwsDoc::EC2::TerminateInstances(instanceID, clientConfig);
    }
    Aws::ShutdownAPI(options);
    return 0;
}

#endif // TESTING_BUILD

File: php/example_code/ec2/TerminateInstances.php
<?php
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

// snippet-start:[ec2.php.terminate_instance.complete]
// snippet-start:[ec2.php.terminate_instance.import]

require 'vendor/autoload.php';

// snippet-end:[ec2.php.terminate_instance.import]
/**
 * Terminate Instances
 *
 * This code expects that you have AWS credentials set up per:
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials.html
 */

// snippet-start:[ec2.php.terminate_instance.main]
$ec2Client = new Aws\Ec2\Ec2Client([
    'region' => 'us-west-2',
    'version' => '2016-11-15',
    'profile' => 'default'
]);

$result = $ec2Client->terminateInstances([
    'DryRun' => true, //true || false,
    // InstanceIds is required
    'InstanceIds' => array('InstanceId1', 'InstanceId2'),
]);

var_dump($result);

// snippet-end:[ec2.php.terminate_instance.main]
// snippet-end:[ec2.php.terminate_instance.complete]
